### PR TITLE
[ENG-3036] feat: Rename accounts from the change account screen

### PR DESCRIPTION
**Pull request by dhriaznov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Enhancement

# 📜 Background
As an Xverse user
I want to see rename my accounts after I’ve selected them
So that I can identify them easily when selecting & switching accounts

Issue Link: #[[ENG-3036](https://linear.app/xverseapp/issue/ENG-3036/%F0%9F%92%B2%F0%9F%8C%8D-rename-accounts-from-the-extensions-change-account-screen)]
Context Link (if applicable):

# 🔄 Changes
- Revamped the Change Account screen to match the new figma design
- Added the possibility to rename ledger and regular accounts
- Added the `validateAccountName` helper func and used it to validate the naming/renaming of ledger and regular accounts 
- Made some code clean-up
- Updated the related locale keys

Impact:
- This PR impacts the Change Account screen
- It also impacts the account naming logic on the ledger account import flow

# 🖼 Screenshot / 📹 Video
Videos are in the linear ticket.

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/screens/accountList/index.tsx [hunk 1]
```diff
@@ -118,33 +92,36 @@ function AccountList(): JSX.Element {
...
-              isSelected={isAccountSelected(account)}
-              onAccountSelected={handleAccountSelect}
-              isAccountListView
-            />
-            <Separator />
-          </div>
-        ))}
-      </AccountContainer>
+      <div>
+        <TopRow title="" onClick={handleBackButtonClick} />
```
Comment (jordankzf) [inline src/app/screens/accountList/index.tsx:96]: Could we consider modifying `TopRow` to make the `title` prop optional?

File: src/app/stores/wallet/reducer.ts [hunk 2]
```diff
@@ -263,6 +264,21 @@ const walletReducer = (
...
+          account.id === action.updatedAccount.id
+            ? action.updatedAccount
+            : account,
+        ),
+        selectedAccount:
+          state.selectedAccount?.accountType === action.updatedAccount.accountType &&
+          state.selectedAccount?.id === action.updatedAccount.id
+            ? { ...state.selectedAccount, accountName: action.updatedAccount.accountName }
+            : state.selectedAccount,
+      };
```
Comment (fedeerbes) [inline src/app/stores/wallet/reducer.ts:281]: question: why not adding this logic in useWalletReducer?

**Review by fedeerbes — DISMISSED** [review]
LGTM, left a question and there is also @jordankzf suggestion.

File: src/app/screens/accountList/index.tsx (hunk 1 above)
Comment (dhriaznov) [reply src/app/screens/accountList/index.tsx:96]: Makes sense to me, I'll make this change

File: src/app/stores/wallet/reducer.ts (hunk 2 above)
Comment (dhriaznov) [reply src/app/stores/wallet/reducer.ts]: good call! updated this logic

**Comment by dhriaznov** [comment]
Thanks for the review folks, I updated the PR, feel free to take a second look 🙌

**Review by fedeerbes — DISMISSED** [review]
LGTM 🚀

**Review by jordankzf — APPROVED** [review]

**Comment by dhriaznov** [comment]
I just resolved the git conflicts, could you approve again pls? @fedeerbes

**Review by fedeerbes — APPROVED** [review]

**Comment by dhriaznov** [comment]
Fixed the issue Dušana found, @fedeerbes could you please review?

**Review by fedeerbes — APPROVED** [review]

**Comment by DuskaT021** [comment]
tested can be merged 👌

**Pull request merged**